### DISCUSSION TITLE
Retrieve user from token

### DISCUSSION THREAD

**Discussion by regisin** [pocketbase/pocketbase#1077] [Q&A]
Is there a way I can retrieve the user given a bearer token?

**Comment by ganigeorgiev** [comment]
The "Auth Refresh" action can do that (it will return the user related to the `Authorization` token and a new token with updated exp claim):

```
Authorization: TOKEN
POST /api/collections/users/auth-refresh
```
(https://pocketbase.io/docs/api-records/#auth-record-actions)

If you are using the SDKs, you can set a `Authorization` header by updating the `authStore` state:

```js
const pb = new PocketBase("https://127.0.0.1:8090");

pb.authStore.save("TOKEN", null);

const { record, token } = await pb.collection("users").authRefresh();
```

**Reply by ganigeorgiev** [reply]
Of course, you can also just sent a request to the "View record" action, but you need to satisfy the "View API rule" of the users collection and extract the record id from the token payload upfront (it is base64 encoded as the second part of the JWT).

**Reply by regisin** [reply]
Thanks. I'm not using any SDK. I'm not using any fancy front end framework, so all interactions with PB will be done by an intermediary backend. The user will never directly interact with the PB "backbackend".

I'll try the first solution just to see if I can manage it. But I think the second one would be better since it wouldn't trigger a new token for each request. I'll just need to learn how to extract the id from the token. Not my forte.

**Reply by ganigeorgiev** [reply]
I wouldn't worry too much about generating new tokens. PocketBase is completely stateless and we don't store the tokens in the database so there are no additional tidy/cleanup scripts to keep for example some tokens table not to grow too big, if that is your concern. The JWT tokens are signed with HS256 which is fairly performant and I haven't experienced any noticeable slow-downs while the application is running.
